compiler-utils/src/testFixtures/java/com/sqareup/anvil/compiler/internal/testing/TestUtils.kt

```diff
@@ -0,0 +1,181 @@
+ */
+@ExperimentalAnvilApi
+public fun compileAnvil(
+  vararg sources: String,
+  enableDaggerAnnotationProcessor: Boolean = false,
+  generateDaggerFactories: Boolean = false,
+  generateDaggerFactoriesOnly: Boolean = false,
+  disableComponentMerging: Boolean = false,
+  allWarningsAsErrors: Boolean = true,
+  useIR: Boolean = true,
+  messageOutputStream: OutputStream = System.out,
+  block: Result.() -> Unit = { }
+): Result {
+  return KotlinCompilation()
+    .apply {
+      compilerPlugins = listOf(AnvilComponentRegistrar())
+      this.useIR = useIR
+      useOldBackend = !useIR
+      inheritClassPath = true
+      jvmTarget = JvmTarget.JVM_1_8.description
+      verbose = false
+      this.allWarningsAsErrors = allWarningsAsErrors
+      this.messageOutputStream = messageOutputStream
+
+      if (enableDaggerAnnotationProcessor) {
+        annotationProcessors = listOf(ComponentProcessor())
+      }
+
+      val anvilCommandLineProcessor = AnvilCommandLineProcessor()
+      commandLineProcessors = listOf(anvilCommandLineProcessor)
+
+      pluginOptions = listOf(
+        PluginOption(
+          pluginId = anvilCommandLineProcessor.pluginId,
+          optionName = "src-gen-dir",
+          optionValue = File(workingDir, "build/anvil").absolutePath
+        ),
+        PluginOption(
+          pluginId = anvilCommandLineProcessor.pluginId,
+          optionName = "generate-dagger-factories",
+          optionValue = generateDaggerFactories.toString()
+        ),
+        PluginOption(
+          pluginId = anvilCommandLineProcessor.pluginId,
+          optionName = "generate-dagger-factories-only",
+          optionValue = generateDaggerFactoriesOnly.toString()
+        ),
+        PluginOption(
+          pluginId = anvilCommandLineProcessor.pluginId,
+          optionName = "disable-component-merging",
+          optionValue = disableComponentMerging.toString()
+        )
+      )
+
+      this.sources = sources.map { content ->
+        val packageDir = content.lines()
+          .firstOrNull { it.trim().startsWith("package ") }
+          ?.substringAfter("package ")
+          ?.replace('.', '/')
+          ?.let { "$it/" }
+          ?: ""
+
+        val name = "${workingDir.absolutePath}/sources/src/main/java/$packageDir/Source.kt"
+        with(File(name).parentFile) {
+          check(exists() || mkdirs())
+        }
+
+        SourceFile.kotlin(name, contents = content, trimIndent = true)
+      }
+    }
+    .compile()
+    .also(block)
+}
+
+@ExperimentalAnvilApi
+public fun Class<*>.moduleFactoryClass(
+  providerMethodName: String,
+  companion: Boolean = false
+): Class<*> {
+  val companionString = if (companion) "_Companion" else ""
+  val enclosingClassString = enclosingClass?.let { "${it.simpleName}_" } ?: ""
+
+  return classLoader.loadClass(
+    "${packageName()}$enclosingClassString$simpleName$companionString" +
+      "_${providerMethodName.capitalize(Locale.US)}Factory"
```

Comment:
Didn't you have an extension override for this one?

Reply:
Yes, that's the Kotlin 1.5 PR that I still did not merge. But I rebased it already on top of this one (which was easier to resolve for merge conflicts) and I'll address there.